client/src/search/Typeahead/TypeaheadMenu.js

```diff
@@ -0,0 +1,114 @@
+import { ListGroup, ListGroupItem } from "react-bootstrap";
+
+import { TM } from "../../components/TextMaker.js";
+
+import { create_text_maker } from "../../models/text.js";
+
+import text from "./Typeahead.yaml";
+
+const text_maker = create_text_maker(text);
+const TextMaker = (props) => <TM tmf={text_maker} {...props} />;
+
+export class TypeaheadMenu extends React.Component {
+  handleWindowClick = (e) => {
+    const { hide_menu } = this.props;
+    if (!document.getElementsByClassName("rbt")[0].contains(e.target)) {
+      hide_menu();
+    }
+  };
+
+  render() {
+    const {
+      search_text,
+      queried_results,
+      config_groups,
+      on_select_item,
+      hide_menu,
+    } = this.props;
+
+    const menu_item_selected = (selected) => {
+      on_select_item(selected);
+    };
+
+    const total_matching_results = queried_results.length;
+
+    if (_.isEmpty(queried_results)) {
+      return (
+        <ListGroup className="rbt-menu dropdown-menu show">
+          <ListGroupItem disabled className="dropdown-item">
+            {text_maker("no_matches_found")}
+          </ListGroupItem>
+        </ListGroup>
+      );
+    } else {
+      return (
+        <ListGroup className="rbt-menu dropdown-menu show">
+          {_.chain(queried_results)
+            .groupBy("config_group_index")
+            .thru((grouped_results) => {
+              let index_key_counter = 0;
```

Comment:
Is this one necessary? `_.uniqueId` or just getting counter from loop seem good enough?